src/darwin/Framework/CHIP/MTRAsyncWorkQueue.mm

```diff
@@ -29,6 +30,20 @@ typedef NS_ENUM(NSInteger, MTRAsyncWorkItemState) {
     MTRAsyncWorkItemRetryCountBase = MTRAsyncWorkItemRunning, // values >= MTRAsyncWorkItemRunning encode retryCount
 };
 
+// A helper struct that facilitates access to _context while
+//  - only reading the _context weak reference once and retaining a strong
+//    reference for the duration of a particular queue method call
+//  - avoiding calls to `[context description]` under our lock
+struct ContextSnapshot {
+    id _Nullable reference;
+    NSString * description;
+    ContextSnapshot(id _Nullable context)
+    {
+        reference = context;
+        description = [context description] ?: @"?";
```

Comment:
I feel like this could just be `description = [context description];` because logging nil with a "%@" format results in the default "(null)" which is easily understandable.

Reply:
Done